pulp_ansible/tests/functional/api/collection/v3/test_collection.py

```diff
@@ -0,0 +1,281 @@
+# coding=utf-8
+"""Tests related to upload of collections."""
+import hashlib
+import logging
+import re
+from urllib.parse import urljoin
+
+import pytest
+
+from pulp_smash import api, config
+from pulp_smash.pulp3.constants import REPO_PATH
+from pulp_smash.pulp3.utils import gen_distribution, gen_repo
+from pulp_smash.utils import http_get
+
+from requests.exceptions import HTTPError
+
+from pulp_ansible.tests.functional.constants import (
+    ANSIBLE_COLLECTION_FILE_NAME,
+    ANSIBLE_COLLECTION_UPLOAD_FIXTURE_URL,
+    ANSIBLE_DISTRIBUTION_PATH,
+    COLLECTION_METADATA,
+)
+from pulp_ansible.tests.functional.utils import set_up_module as setUpModule  # noqa:F401
+
+from orionutils.generator import build_collection
+
+
+logger = logging.getLogger(__name__)
+
+
+def upload_handler(client, response):
```

Comment:
This one could be added to pulp-smash itself. if we think other tests will re-use this.